### PR TITLE
tests: add tests and fix typo

### PR DESCRIPTION
**Pull request by Crowley723**
## Summary by CodeRabbit

* **Bug Fixes**
  * Duration parsing now surfaces numeric parse errors instead of silently accepting invalid values, preventing unexpected behavior with large, negative, or malformed inputs.

* **Tests**
  * Added extensive unit tests covering edge cases (overflow, bounds, leading zeros, zero/max values) to ensure consistent error propagation and parsing behavior.

### CODE REVIEW

**Review by james-d-elliott — APPROVED** [review]

File: internal/utils/time_test.go [hunk 1]
```diff
@@ -262,3 +262,217 @@
...
+			name:     "LargeButValid64BitNumber",
+			input:    "9223372036854775807s", // Max int64
+			expected: "9223372036854775807s",
+		},
+		{
+			name:     "MultipleValidLargeNumbers",
+			input:    "2147483647h2147483647m2147483647s",
+			expected: "2147483647h2147483647m2147483647s",
+		},
+	}
```
[the remark this answers is not in the document]
Comment (Crowley723) [reply internal/utils/time_test.go:447]: @james-d-elliott Do we care about this?

[the remark this answers is not in the document]
Comment (james-d-elliott) [reply internal/utils/time_test.go:447]: Yeah we can ignore this.

**Review by autheliabot — APPROVED** [review]
